src/app/core/components/language-picker/language-picker.component.ts

```diff
@@ -1,28 +1,31 @@
-import { Component, Input, OnChanges, Output, EventEmitter } from '@angular/core';
+import { Component, Input, OnChanges, Output, EventEmitter, OnInit } from '@angular/core';
+import { LanguageConfig } from 'src/app/shared/helpers/config';
 import { LocaleService } from '../../services/localeService';
 
 @Component({
   selector: 'alg-language-picker',
   templateUrl: './language-picker.component.html',
   styleUrls: [ './language-picker.component.scss' ]
 })
-export class LanguagePickerComponent implements OnChanges {
+export class LanguagePickerComponent implements OnInit, OnChanges {
   @Input() styleClass?: string;
   @Input() defaultLang?: string;
   @Input() redirectOnChange = true;
   @Output() changeLang = new EventEmitter<string>();
 
   readonly languages = this.localeService.languages;
-  current = this.localeService.currentLang;
+  current?: LanguageConfig
 
   constructor(
     private localeService: LocaleService,
   ) {}
 
+  ngOnInit(): void {
+    this.initCurrentLang();
+  }
+
   ngOnChanges(): void {
-    if (this.defaultLang) {
-      this.current = this.languages.find(l => l.tag === this.defaultLang);
-    }
+    this.initCurrentLang();
```

Comment:
ngChanges is called at init as well (check it), so no need to have ngInit and a separated function

Reply:
I checked it before adding the `ngOnInit` method, the `ngOnChanges` is not triggered at init